hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/StructuredTraceGraphBuilder.java

```diff
@@ -0,0 +1,47 @@
+package org.hypertrace.traceenricher.trace.util;
+
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.core.datamodel.shared.StructuredTraceGraph;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StructuredTraceGraphBuilder {
+  private static final Logger LOG = LoggerFactory.getLogger(StructuredTraceGraphBuilder.class);
+
+  private static ThreadLocal<StructuredTraceGraph> cachedGraph = new ThreadLocal<>();
+  private static ThreadLocal<StructuredTrace> cachedTrace = new ThreadLocal<>();
+
+  public static StructuredTraceGraph buildGraph(StructuredTrace trace) {
```

Comment:
@kotharironak  Basic question, Why buildGraph is called for every span. Can we call only once for Trace instead of every span ?

Reply:
@pavan-traceable Where does the buildGraph get called for every span?

Reply:
Thats my understanding from this slack chat

Reply:
@pavan-traceable buildGraph is called at 3 places, 
ApiBoundaryTypeAttributeEnricher
DefaultServiceEntityEnricher
BackendEntityEnricher
Also we work with complete trace in this service, so buildGraph is invoked at Trace level